extensions/ql-vscode/src/view/compare-performance/ComparePerformance.tsx

```diff
@@ -860,3 +881,14 @@ function collatePipelines(
 function samePipeline(a: string[], b: string[]) {
   return a.length === b.length && a.every((x, i) => x === b[i]);
 }
+
+function getPipelineSummaryHash(pipelines: Record<string, PipelineSummary>) {
+  // Note: we can't import "crypto" here because it is not available in the browser,
+  // so we just concatenate the hashes of the individual pipelines.
```

Comment:
How long will this be? Can we possibly have 100s of keys? It might be safer to do some checks to ensure that the `result` is never larger than a fixed amount. And if it is, then you can sample the first `n` chars in each key to keep it under the limit, retain a reasonable amount of stochasticity, and ensure the result easy to reproduce given the same input.

Reply:
Back when the "standard order" existed there could be multiple pipelines, but nowadays there are 2 for recursive predicates and 1 for all other predicates. So it's really small.